### PR TITLE
Add replace operator

### LINKED ISSUE #895: [ENH] Support SQL "replace" function
```
>>> from dask_sql import Context
>>> import pandas as pd

>>> c = Context()
>>> df = pd.DataFrame({"id": ["abc", "def", "bcd"]})
>>> c.create_table("df", df)
>>> c.sql("select replace(id, 'b', '') from df").compute()
Traceback (most recent call last):
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/physical/rex/core/call.py", line 1004, in convert
    operation = self.OPERATION_MAPPING[operator_name]
KeyError: 'replace'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/physical/rex/core/call.py", line 1007, in convert
    operation = context.schema[schema_name].functions[operator_name]
KeyError: 'replace'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/context.py", line 501, in sql
    return self._compute_table_from_rel(rel, return_futures)
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/context.py", line 830, in _compute_table_from_rel
    dc = RelConverter.convert(rel, context=self)
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/physical/rel/convert.py", line 61, in convert
    df = plugin_instance.convert(rel, context=context)
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/physical/rel/logical/project.py", line 57, in convert
    new_columns[random_name] = RexConverter.convert(
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/physical/rex/convert.py", line 73, in convert
    df = plugin_instance.convert(rel, rex, dc, context=context)
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/physical/rex/core/call.py", line 1009, in convert
    raise NotImplementedError(f"{operator_name} not (yet) implemented")
NotImplementedError: replace not (yet) implemented
```

Expected:
```
ac
def
cd
```

### PR DESCRIPTION
**Pull request by ChrisJar**
Closes #895

### CODE REVIEW

File: tests/integration/test_rex.py [hunk 1]
```diff
@@ -522,7 +522,9 @@ def test_string_functions(c, gpu):
             SUBSTR(a, 3, 6) AS s,
             INITCAP(a) AS t,
             INITCAP(UPPER(a)) AS u,
-            INITCAP(LOWER(a)) AS v
+            INITCAP(LOWER(a)) AS v,
+            REPLACE(a, 'r', 'l') as w,
+            REPLACE('Another String', 'th', 'b') as x
```
Comment (ayushdg) [inline tests/integration/test_rex.py:527]: Could you also add some tests to the `unit/test_call::test_string_operations` unit tests. Preferable with some edge cases if possible

**Review by ayushdg — APPROVED** [review]
